src/components/trade/TradesTable/index.tsx

```diff
@@ -0,0 +1,176 @@
+  const tableContext = React.useContext(TradesTableContext)
+
+  return (
+    <tr>
+      <th>Order Id</th>
+      <th>Type</th>
+      <th>Sell Amount</th>
+      <th>Buy Amount</th>
+      <th>
+        Limit price
+        <Icon icon={faExchangeAlt} onClick={(): void => tableContext.inverPrice()} />
+      </th>
+      <th>
+        Execution Price
+        <Icon icon={faExchangeAlt} onClick={(): void => tableContext.inverPrice()} />
+      </th>
+      <th>Tx Hash</th>
+      <th>Surplus</th>
+      <th>Trade Time</th>
+    </tr>
+  )
+}
+
+export type Props = {
+  header?: JSX.Element
+  className?: string
+  numColumns?: number
+  data: Array<Order>
+}
+
+const TitleUid = styled(RowWithCopyButton)`
+  color: ${({ theme }): string => theme.grey};
+  font-size: ${({ theme }): string => theme.fontSizeDefault};
+  font-weight: ${({ theme }): string => theme.fontNormal};
+  display: flex;
+  align-items: center;
+`
+
+export const TradesTable = ({ header, className, numColumns, data }: Props): JSX.Element => {
+  const tableContext = React.useContext(TradesTableContext)
+  const [rows, setRows] = React.useState(<></>)
+
+  React.useEffect((): void => {
+    // for each entry in the data array invert the values of
+    // the limit price and execution price if toggled
+    const _rows = data.map((order, i): JSX.Element => {
+      interface price {
+        amount: BigNumber
+        fraction: TokenErc20
+      }
+      interface priceData {
+        denominator: price
+        numerator: price
+      }
+
+      const constructPrice = (data: priceData): string => {
+        const calculatedPrice = calculatePrice({
+          denominator: { amount: data.denominator.amount, decimals: data.denominator.fraction?.decimals },
+          numerator: { amount: data.numerator.amount, decimals: data.numerator.fraction?.decimals },
+        })
+
+        const displayPrice = (tableContext.isPriceInverted ? invertPrice(calculatedPrice) : calculatedPrice).toString(
+          10,
+        )
+        const formattedPrice = formatSmart({
+          amount: displayPrice,
+          precision: NO_ADJUSTMENT_NEEDED_PRECISION,
+          smallLimit: HIGH_PRECISION_SMALL_LIMIT,
+          decimals: HIGH_PRECISION_DECIMALS,
+        })
+        const buySymbol = order.buyToken ? safeTokenName(order.buyToken) : ''
+        const sellSymbol = order.sellToken ? safeTokenName(order.sellToken) : ''
+        const [baseSymbol, quoteSymbol] = tableContext.isPriceInverted
+          ? [sellSymbol, buySymbol]
+          : [buySymbol, sellSymbol]
+
+        return `${formattedPrice} ${quoteSymbol} for ${baseSymbol}`
+      }
+
+      return (
+        <tr key={i}>
+          <td>
+            <TitleUid textToCopy={order.uid} contentsToDisplay={order.shortId} />
+          </td>
+          <td className={i % 2 === 1 ? 'long' : 'short'}>{i % 2 === 1 ? 'Buy' : 'Sell'}</td>
+          <td>
+            {order.sellToken?.decimals.toFixed(3)} {order.sellToken?.symbol}
```

Comment:
We want to display the amounts, not the decimals ;)
Also, we'll probably want to use `formatSmart` for the amounts. See how that's done on https://github.com/gnosis/gp-ui/blob/dfb80806a9577f9bbb223bc999fa9fd0e540e60b/src/components/orders/AmountsDisplay/index.tsx#L33